x/stakeibc/keeper/reward_converter.go

```diff
@@ -133,57 +133,56 @@ func (k Keeper) TransferConvertedTokensTradeToHost(ctx sdk.Context, amount sdk.I
 // Trade reward tokens in the Trade ICA for the target output token type using ICA remote tx on trade zone
 // The amount represents the total amount of the reward token in the trade ICA found by the calling ICQ
 // Depending on min and max swap amounts set in the route, it is possible not the full amount given will swap
-func (k Keeper) TradeRewardTokens(ctx sdk.Context, amount sdk.Int, route types.TradeRoute) error {
+func (k Keeper) TradeRewardTokens(ctx sdk.Context, rewardAmount sdk.Int, route types.TradeRoute) error {
 	// If the min swap amount was not set it would be ZeroInt, if positive we need to compare to the amount given
 	//  then if the min swap amount is greater than the current amount, do nothing this epoch to avoid small swaps
-	if route.MinSwapAmount.IsPositive() && route.MinSwapAmount.GT(amount) {
+	if route.MinSwapAmount.IsPositive() && route.MinSwapAmount.GT(rewardAmount) {
 		return nil
 	}
 
 	// If the max swap amount was not set it would be ZeroInt, if positive we need to compare to the amount given
 	//  then if max swap amount is LTE to amount full swap is possible so amount is fine, otherwise set amount to max
-	if route.MaxSwapAmount.IsPositive() && route.MaxSwapAmount.GT(amount) {
-		amount = route.MaxSwapAmount
+	if route.MaxSwapAmount.IsPositive() && route.MaxSwapAmount.GT(rewardAmount) {
+		rewardAmount = route.MaxSwapAmount
```

Comment:
Is this a possible weird race condition: the ICQ detects 1.5x the maxSwapAmount so we fire off a trade command with the amount being maxSwapAmount... afterwards there will be 0.5x maxSwapAmount in the trade ICA so the next swap command wouldn't need to modify it... but what if the swap ICA hasn't executed by the time that the next epoch ICQ on the reward ICA happens? So the ICQ still sees 1.5x maxSwapAmount and so it asks for more to be swapped than will currently be available.  Can we protect against this by making the swap ICA timeout on or slightly before the epoch boundaries?

Reply:
hmm, good question. But I don't think there's a race condition cause will have to have succeeded or failed before the next epoch, and the query isn't submitted until the next epoch

Reply:
Yes that was my thinking as well so currently it should be fine.  If we ever make the ICA controller work in the way occasionally discussed where it stores down the query this  block but doesn't send until the next block, that could re-open this issue on the rare case that the send block falls on the start of a new Stride epoch